ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-content.component.js

```diff
@@ -224,15 +225,11 @@ export default class ConfirmPageContainerContent extends Component {
                       <Button
                         type="inline"
                         className="confirm-page-container-content__link"
-                        onClick={() => {
-                          global.platform.openTab({
-                            url: getPortfolioUrl(
-                              'buy',
-                              'ext_buy_button',
-                              metaMetricsId,
-                            ),
-                          });
-                        }}
+                        onClick={() =>
+                          openBuyCryptoInPdapp({
+                            metaMetricsId,
+                          })
+                        }
```

Comment:
@darkwing This is touching one of your [very recent changes](https://github.com/MetaMask/metamask-extension/commit/e339afce7a421ea9142660dedffc10c804de4128#diff-2b2dbd03c1ef0d393b7442606996ff80b03bfb295461a202e09e7b26344ca2faL227-R233), please review